### DISCUSSION TITLE
pyinstaller packaging problem

### DISCUSSION THREAD

**Discussion by lmcjrrg** [pyinstaller/pyinstaller#7857] [Help]
machine A on my ubuntu system uses pyinstaller for program packaging. there is a complete operating environment for the program on machine A. I run the packaged binary execution file on this machine, and it can run normally, but upload this binary file to a new on ubuntu machine B, machine B does not have a program running environment. if you run it again, an error will be reported. will the executable file packaged by pyinstaller still depend on the environment other than the decompressed _MEI* folder?

**Comment by lmcjrrg** [comment]
this is the .spec file packaged by pyinstaller:
[start_program.txt](https://github.com/pyinstaller/pyinstaller/files/12365692/start_program.txt)
package log:
[packaged_log.txt](https://github.com/pyinstaller/pyinstaller/files/12365688/packaged_log.txt)
execution error:
![image](https://github.com/pyinstaller/pyinstaller/assets/95133674/8035760d-e918-47bd-9e77-11e8ca54f161)

**Comment by rokm** [comment]
This looks like your `init_support_multi_rtsp` is trying to load a shared library at fixed location - so that naturally will not work unless you have your deepstream libraries installed at `/opt/nvidia/deepstream/deepstream-6.1/lib`. Alternatively, it could be that one of extensions that you load (or its dependency) is build with rpath that points at that location, so it expects to find the library there.

This
>     binaries=[('pretreatment.so','./'),('support_multi_rtsp.so','./'),('invoke_external_server.so','./'),('/opt/nvidia/deepstream/deepstream-6.1/lib/*.so','/opt/nvidia/deepstream/deepstream-6.1/lib')],

also looks wrong - the destination for collected files should be a directory relative to the application directory, not an absolute path. This should already raise an error in onedir builds, but in onefile build, it probably just quietly creates the files in `_MEI**/opt/nvidia/deepstream/deepstream-6.1/lib`.

**Reply by lmcjrrg** [reply]
thanks for the reply, well, i installed the deepstream library on the machine /opt/nvidia/deepstream/deepstream-6.1/lib, the program will report other dependency errors, and the dependencies seem to be a bit complicated, so you mean, I can only in Is the deepstream runtime environment also installed on the new machine?

**Reply by rokm** [reply]
> so you mean, I can only in Is the deepstream runtime environment also installed on the new machine?

I don't know - it depends on why whatever requires `libnvds_meta.so` is looking for it in `/opt/nvidia...`, and whether you can modify that to try to load the copy you bundle with your application.

**Reply by lmcjrrg** [reply]
ok, let me explore again, anyway, thank you for your reply
